src/parser/type/datetime/date_type_std.cpp

```diff
@@ -0,0 +1,214 @@
+            continue;
+        }
+        if (std::isdigit(date_ptr[pos])) {
+            year = year * 10 + (date_ptr[pos] - '0');
+            ++pos;
+            // if (year > 9999)
+            //     return false;
+            continue;
+        }
+        break;
+    }
+    if (negative_year) {
+        year = -year;
+    }
+    if (date_ptr[pos] != '-' && date_ptr[pos] != '/') {
+        return false;
+    }
+    ++pos; // skip - and /
+
+    // Get month
+    int32_t month{0};
+    while (pos < length && std::isdigit(date_ptr[pos])) {
+        if (std::isspace(date_ptr[pos])) {
+            ++pos;
+            continue;
+        }
+        if (std::isdigit(date_ptr[pos])) {
+            month = month * 10 + (date_ptr[pos] - '0');
+            ++pos;
+            // if (month > 12)
+            //     return false;
+            continue;
+        }
+        break;
+    }
+    if (date_ptr[pos] != '-' && date_ptr[pos] != '/') {
+        return false;
+    }
+    ++pos; // skip - and /
+
+    // Get day
+    int32_t day{0};
+    while (pos < length && std::isdigit(date_ptr[pos])) {
+        if (std::isspace(date_ptr[pos])) {
+            ++pos;
+            continue;
+        }
+        if (std::isdigit(date_ptr[pos])) {
+            day = day * 10 + (date_ptr[pos] - '0');
+            ++pos;
+            // if (day > 31)
+            //     return false;
+            continue;
+        }
+        break;
+    }
+    end_length = pos;
+
+    return YMD2Date(year, month, day, date);
+}
+
+bool DateTypeStd::YMD2Date(int32_t year, int32_t month, int32_t day, DateTypeStd &date) { 
+    std::chrono::year_month_day ymd (
+        std::chrono::year{year},
+        std::chrono::month{static_cast<unsigned int>(month)},
+        std::chrono::day{static_cast<unsigned int>(day)}
+    );
+    auto sd = std::chrono::sys_days{ymd};
+    date.value = sd.time_since_epoch().count();
+    return ymd.ok();
+}
+
+bool DateTypeStd::Date2YMD(int32_t days, int32_t &year, int32_t &month, int32_t &day) {
+    auto sd = std::chrono::sys_days{std::chrono::days{days}};
+    auto ymd = std::chrono::year_month_day{sd};
+
+    year = static_cast<int>(ymd.year());
+    month = static_cast<unsigned>(ymd.month());
+    day = static_cast<unsigned>(ymd.day());
+    return IsDateValid(year, month, day);
+}
+
+bool DateTypeStd::IsDateValid(int32_t year, int32_t month, int32_t day) {
+    DateTypeStd date;
+    return YMD2Date(year, month, day, date);
+}
+
+bool DateTypeStd::Add(DateTypeStd input, IntervalType interval, DateTypeStd &output) {
+    int32_t year{0}, month{0}, day{0};
+    if (!Date2YMD(input.value, year, month, day)) {
+        return false;
+    }
+    switch (interval.unit) {
+        case kYear: {
+            std::chrono::sys_days output_sd(std::chrono::days{input.value} + std::chrono::duration_cast<std::chrono::days>(std::chrono::years{interval.value}));
+            output.value = output_sd.time_since_epoch().count();
+            return true;
+        }
```

Comment:
for year and month, check https://en.cppreference.com/w/cpp/chrono/year_month_day/operator_arith

Reply:
Just to clarify, when using std::chrono::year_month_day, "2020-1-30" + 1 month will yield "2020-2-30", then it will be adjusted to "2020-03-01", is this behavior desired? It seems that the original pull request matches the functionalities of previously existing type DateT.